src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -69,15 +74,65 @@ public struct MetricPoint
         /// </summary>
         public ReadOnlyTagCollection Tags { get; }
 
-        public DateTimeOffset StartTime { get; internal set; }
+        internal MetricPointStatus MetricPointStatus { get; private set; }
 
-        public DateTimeOffset EndTime { get; internal set; }
+        public DateTimeOffset GetStartTime()
+        {
+            return this.StartTime;
+        }
 
-        public long LongValue { get; internal set; }
+        public DateTimeOffset GetEndTime()
+        {
+            return this.EndTime;
+        }
 
-        public double DoubleValue { get; internal set; }
+        public long GetLongSum()
+        {
+            if (this.aggType == AggregationType.LongSumIncomingDelta || this.aggType == AggregationType.LongSumIncomingCumulative)
+            {
+                return this.longValue;
+            }
+            else
+            {
+                throw new NotSupportedException($"{nameof(this.GetLongSum)} is not supported for this metric type.");
+            }
+        }
 
-        internal MetricPointStatus MetricPointStatus { get; private set; }
+        public double GetDoubleSum()
```

Comment:
```suggestion
        public double GetSumDouble()
```

^ thoughts?

Reply:
GetSum<Double> 😆

Reply:
To me `GetDoubleSum` reads better than `GetSumDouble`. 
But:  below I think it makes sense to call `GetLastValueLong` as others suggested. So.. now I'm in doubt on this one. It would be kinda inconsistent then if we go with `GetDoubleSum`, no? 🤔

Reply:
@joaopgrassi https://github.com/open-telemetry/opentelemetry-dotnet/pull/2667#discussion_r756334509 we currently settled with this. Please let us know what do you think.